lua/nvim-treesitter/configs.lua

```diff
@@ -1,88 +1,327 @@
-  },
-  java = {
+  }
+}
+
+parsers.java = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-java",
     files = { "src/parser.c" },
-  },
-  html = {
+  }
+}
+
+parsers.html = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-html",
     files = { "src/parser.c", "src/scanner.cc" },
-  },
-  julia = {
+  }
+}
+
+parsers.julia = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-julia",
     files = { "src/parser.c", "src/scanner.c" },
-  },
-  json = {
+  }
+}
+
+parsers.json = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-json",
     files = { "src/parser.c" },
-  },
-  css = {
+  }
+}
+
+parsers.css = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-css",
     files = { "src/parser.c", "src/scanner.c" },
-  },
-  ocaml = {
+  }
+}
+
+parsers.ocaml = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-ocaml",
     files = { "src/parser.c", "src/scanner.cc" },
-  },
-  swift = {
+  }
+}
+
+parsers.swift = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-swift",
     files = { "src/parser.c" },
-  },
-  csharp = {
+  }
+}
+
+parsers.csharp = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-c-sharp",
     files = { "src/parser.c", "src/scanner.c" },
-  },
-  typescript = {
+  }
+}
+
+parsers.typescript = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-typescript",
     files = { "src/parser.c", "src/scanner.c" },
     location = "tree-sitter-typescript/typescript"
-  },
-  tsx = {
+  }
+}
+
+parsers.tsx = {
+  install_info = {
     url = "https://github.com/tree-sitter/tree-sitter-typescript",
     files = { "src/parser.c", "src/scanner.c" },
     location = "tree-sitter-tsx/tsx"
   }
 }
 
+-- @enable can be true or false
+-- @disable is a list of languages, only relevant if enable is true
+-- @keymaps list of user mappings for a given module if relevant
+-- @is_enabled function which, given a ft, will return true if the ft works on the module
+local config = {
+  highlight = {
+    enable = false,
+    disable = {},
+    is_enabled = function(ft)
+      return queries.get_query(ft, 'highlights') ~= nil
+    end
+  },
+  -- selection = {
+  --   enable = false,
+  --   disable = {},
+  --   keymaps = {},
+  --   is_enabled = function() return false end
+  -- },
+  -- folding = {
+  --   enable = false,
+  --   disable = {},
+  --   keymaps = {},
+  --   is_enabled = function() return false end
+  -- }
+}
+
+local M = {}
+
+local function enable_module(mod, bufnr, ft)
+  local bufnr = bufnr or api.nvim_get_current_buf()
+  local ft = ft or api.nvim_buf_get_option(bufnr, 'ft')
+  if not parsers[ft] or not config[mod] then
+    return
+  end
+
+  local str = "lua require'nvim-treesitter.%s'.attach(%s, %s)"
+  local cmd = string.format(str, mod, bufnr, ft)
+  api.nvim_command(cmd)
```

Comment:
Sounds quite convoluted here, why not phrasing it like so :
```suggestion
  local loaded_mod = require(string.format('nvim-treesitter.%s', mod))
  local cmd =  loaded_mod.attach(bufnr, ft)
```

Reply:
hum didn't know you could `require` like this. thanks for the tip